src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -639,7 +645,22 @@ public static IDisposable ActionScope(this ILogger logger, ActionDescriptor acti
 
         public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
         {
-            _actionExecuting(logger, action.DisplayName, null);
+            var routeKeys = action.RouteValues.Keys.ToArray();
```

Comment:
Surround all of this work with a call to `IsEnabled` so that you don't pay the cost if logging is off.